unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/WorldChatWindowHUD/PublicChatChannelController.cs

```diff
@@ -221,6 +232,32 @@ private void HandleMessageReceived(ChatMessage message)
 
         if (message.messageType == ChatMessage.Type.PRIVATE && message.recipient == ownProfile.userId)
             lastPrivateMessageRecipient = userProfileBridge.Get(message.sender).userName;
+        
+        deactivatePreviewCancellationToken.Cancel();
+        deactivatePreviewCancellationToken = new CancellationTokenSource();
+        deactivatePreviewHUDControllerCancellationToken.Cancel();
+        deactivatePreviewHUDControllerCancellationToken = new CancellationTokenSource();
+        
+        if (_currentState.Equals(ChatWindowVisualState.INPUT_MODE))
+        {
+            //Input window is present. Start counter to deactivate
+            if (chatHudController.IsInputSelected) return;
+            WaitThenActivatePreview(deactivatePreviewCancellationToken.Token).Forget();
+        }
+        else
+        {
+            //If the chatHudController is visible, just restart the counter
+            if (_currentState.Equals(ChatWindowVisualState.PREVIEW_MODE))
+            {
+                //Start Fade Out Counter
+                WaitThenDeactivatePreviewChatHUDController(deactivatePreviewHUDControllerCancellationToken.Token).Forget();
+            }
+            else
```

Comment:
i would specify the case: `else if (currentState == ChatWindowVisualState.NONE_VISIBLE)` its easier to follow